src/components/SpiceList.js

```diff
@@ -8,7 +8,23 @@ class SpiceList extends React.Component {
     search: ""
   }
 
+//handles the returned search state from the callback ...
+  handleSearchState = (newSearchState) => {
+    this.setState({
+      search: newSearchState
+    })
+  }
+
+  //handles the returned four star checkbox state from the the callback that we sent to the child named Filter. I'm wondering if anyone named their child Filter haha
+  handleCheckedState = (newCheckedState) => {
+    this.setState({
+      fourStarOnly: newCheckedState
+    })
+  }
```

Comment:
Nice! One trick here you could try for shortening the syntax would be to use [shorthand property names](https://ui.dev/shorthand-properties/#:~:text=With%20Shorthand%20Properties%2C%20whenever%20you,can%20omit%20the%20property%20name.) like this:

```js
handleSearchState = search => {
    this.setState({ search })
  }

  handleCheckedState = fourStarOnly => {
    this.setState({ fourStarOnly })
  }
```

This works as long as the key in state matches the name of the variable you're trying to assign to it.

Reply:
This is so neat , thank you!